### PR TITLE
I flubbed the commit on the last one, this one should be good (I hope)

### PR DESCRIPTION
**Pull request by brianb**
This patch replaces the previous one and includes the unit test changes

### CODE REVIEW

File: src/main/java/com/xiaoerge/x12/message/MessageFormat.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+public class MessageFormat {
+    private String dataElementSeparator = "*";
+    private String componentElementSeparator = "\\";
+    private String segmentTerminator = "~";
+    private String repetitionSeparator = "^";
+    private String suffix = null;
+    
+    public MessageFormat() {
+	}
+    public MessageFormat(String s) throws MessageFormatException {
```
Comment (xiaoerge) [inline src/main/java/com/xiaoerge/x12/message/MessageFormat.java]: What do you think about removing this constructor? For couple reasons,  
A) I prefer the parser not throwing hard exception on incorrect message format. Since Segment.java and util classes are only logging warnings. 
B) This constructor method isn't tested for message parsing.

I'd say we remove it and expose setters for them.
The parser requires some knowledge of the incoming message anyway so knowing the proper terminators to set for message format should be fine.

File: src/main/java/com/xiaoerge/x12/message/MessageFormat.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package com.xiaoerge.x12.message;
+
+import java.util.regex.Pattern;
+
+public class MessageFormat {
+    private String dataElementSeparator = "*";
+    private String componentElementSeparator = "\\";
+    private String segmentTerminator = "~";
+    private String repetitionSeparator = "^";
+    private String suffix = null;
```
Comment (xiaoerge) [inline src/main/java/com/xiaoerge/x12/message/MessageFormat.java]: Test this case
http://stackoverflow.com/questions/4260723/concatenating-null-strings-in-java

File: src/main/java/com/xiaoerge/x12/message/MessageFormatException.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package com.xiaoerge.x12.message;
+
+public class MessageFormatException extends Exception {
```
Comment (xiaoerge) [inline src/main/java/com/xiaoerge/x12/message/MessageFormatException.java]: If removing the parametrized constructor for MessageFormat.java,  remove this class as well.

File: src/main/java/com/xiaoerge/x12/message/MessageFormat.java (hunk 2 above)
Comment (brianb) [inline src/main/java/com/xiaoerge/x12/message/MessageFormat.java]: So, the only place suffix is used is in StringQueue and it is checked for null, so we do no string concatenation with null.  I am comfortable using empty string instead.

File: src/main/java/com/xiaoerge/x12/message/MessageFormat.java (hunk 1 above)
Comment (brianb) [inline src/main/java/com/xiaoerge/x12/message/MessageFormat.java]: So the thing is, that the delimiters should be fully discoverable from the ISA segment itself without any advanced knowledge of the message format.  I'd be ok moving this to an auxiliary class, but I wouldn't want to lose the ability to detect the delimiters automagically.

File: src/main/java/com/xiaoerge/x12/message/MessageFormat.java (hunk 2 above)
Comment (xiaoerge) [inline src/main/java/com/xiaoerge/x12/message/MessageFormat.java]: Cool,  lets go with a default empty string.

**Comment by xiaoerge** [comment]
Test the parametrized MessageFormat constructor and it should be all good.

File: src/main/java/com/xiaoerge/x12/message/MessageFormat.java (hunk 2 above)
Comment (brianb) [inline src/main/java/com/xiaoerge/x12/message/MessageFormat.java]: done

**Comment by brianb** [comment]
So, you're saying keep the parameterized constructor, but add a unit test to the patch?

I have another patch in the queue for length validation when the repetition character is present but it depends on these changes.

**Comment by xiaoerge** [comment]
Yes, I agree with your point of having that constructor.  We now need to unit test that case by constructing non default MessageFormat("ISA....").  Try more than one, get the edge cases too.

**Comment by brianb** [comment]
Had a re-think on the parameterized constructor.  I think your initial instinct is correct.

I moved the ISA parsing logic to X12Parser.parseMessageFormat(), added setters to MessageFormat, removed the constructor, and the MessageFormatException class and added some unit testing in X12MessageBaseTest.

**Comment by xiaoerge** [comment]
Thanks @brianb
